### PR TITLE
new functions

### PR DESCRIPTION
**Pull request by Cofflag**

### CODE REVIEW

File: include/attribute.h [hunk 1]
```diff
@@ -270,6 +256,7 @@ struct SaveBlock1
     u8 filler5C8[0xE];
     u16 fishingPoint;
 }; /* size = 0x5D8 */
+/* TODO: list offsets */
```
Comment (jiangzhengwenjz) [inline include/attribute.h:259]: General advice: it's weird to add a comment below. 
Usually we do it above what you want to comment on.

File: src/attribute.c [hunk 2]
```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
...
+    u16 i = 0;
+
+    sb1->unk16[0] = buffer[0];
+    while (buffer[i])
+    {
+        ++i;
+        sb1->unk16[i] = buffer[i];
+    }
+}
+/* The structure of this func is the same to the upper one. But after checking the memory, unk16 is not where the name array lays. */
```
Comment (jiangzhengwenjz) [inline src/attribute.c:38]: Same issue. Please move it above the function. 
And the comment itself is very verbose, you can just say something like `The same as SetPlayerName but uses unk16.` .

File: src/attribute.c [hunk 3]
```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
...
+
+void sub_08017BB4(u8 r0)
+{
+    gSaveBlock1Ptr->playerGender = r0;
+    sub_08012F0C(0x182,r0); // This apprears in main. Maybe I should figure out what this func does.
```
Comment (jiangzhengwenjz) [inline src/attribute.c:43]:
```suggestion
    sub_08012F0C(0x182, r0); // This appears in main. 
```

Comment (jiangzhengwenjz) [inline src/attribute.c:43]: What do you mean by `Maybe I should figure out what this func does.`? We always want to figure out what every function does, right?

File: src/attribute.c [hunk 4]
```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
...
+void sub_08017C20(u32 r0)
+{
+    struct SaveBlock1 *sb1 = gSaveBlock1Ptr;
+    if (r0 > 99999999)
+    {
+        r0 = 99999999;
+    }
+    sb1->playerExp = r0;
+}
+/* The really first one. It keeps showing bhi until I switched the order of if else. Then I realize I can directly use r0 instead of a new variable to remove else. Finally, it matched. */
```
Comment (jiangzhengwenjz) [inline src/attribute.c:82]: Useless comment.

File: src/attribute.c [hunk 5]
```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
...
+}
+
+void sub_08017C88(u8 r0, u16 r1)
+{
+    gSaveBlock1Ptr->playerAglList[r0] = r1;
+}
+
+void sub_08017C9C(u8 r0)
+{
+    gSaveBlock1Ptr->unk58[0] = r0;
```
Comment (jiangzhengwenjz) [inline src/attribute.c:111]: Are you sure `unk58` is an array? (I'm just asking. Not saying you're wrong)

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: src/attribute.c (hunk 3 above)
Comment (Cofflag) [reply src/attribute.c:43]: Yes. I mean maybe I should figure out what this func does recently. ^_^

File: src/attribute.c (hunk 5 above)
Comment (Cofflag) [reply src/attribute.c:111]: I really don't know what this is. I checked this place with my clear save but it keeps showing 00 03 00 00. I tried listing xref to gSaveBlock1Ptr, but it's just too much. I also tried [02001059]?!!, but it only shows that 59 is a byte, I don't know how to figure out what it does. Therefore, I don't know if unk58 is an array or not. T_T

File: src/attribute.c (hunk 3 above)
Comment (jiangzhengwenjz) [reply src/attribute.c]: @Cofflag You missed a space.

File: src/attribute.c [hunk 6]
```diff
@@ -3,6 +3,8 @@
 #include "data.h"
 #include "functions.h"
 
+const u8 gUnk_080BABCA[0xA] = { 1, 2, 3, 4, 5, 6, 7};
```
Comment (jiangzhengwenjz) [inline src/attribute.c:6]:
```suggestion
const u8 gUnk_080BABCA[0xA] = { 1, 2, 3, 4, 5, 6, 7 };
```

File: src/attribute.c [hunk 7]
```diff
@@ -23,3 +25,162 @@ void SetPlayerName(u16 *nameBuffer)
...
+            if (temp <= 0xef)
+            {
+                if (sub_080639E8(temp) == arr1[r5]) 
+                {
+                    r6 = 1;
+                    break;
+                }
+            }
+        }
+
```
Comment (jiangzhengwenjz) [inline src/attribute.c:183]: redundant newline

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

**Review by jiangzhengwenjz — APPROVED** [review]

**Pull request merged**